### ISSUE TITLE
Stuck on ContinueRunningWorkflows

### ISSUE THREAD

**Issue by johnwc** [elsa-workflows/elsa-core#3160]
Our nodes where shutdown while a few workflows were running. Upon startup they all are stuck at `ContinueRunningWorkflows` and will not proceed further. I can not get to the dashboard because it doesn't startup the web server until after all this.

Dashboard node
```
...
info: Elsa.Runtime.StartupRunner[0] Running startup task ContinueRunningWorkflows
info: Elsa.StartupTasks.ContinueRunningWorkflows[0] Found no workflows with status 'Id'. Nothing to resume
info: Elsa.StartupTasks.ContinueRunningWorkflows[0] Found 12 workflows with status 'Running'. Resuming each one of them
```

Worker nodes (6)
```
...
info: Rebus.Threading.TaskParallelLibrary.TplAsyncTask[0] Starting periodic task "CleanupTrackedErrors" with interval 00:00:10
info: Rebus.Bus.RebusBus[0] Bus "Rebus 4" setting number of workers to 1
info: Rebus.Bus.RebusBus[0] Bus "Rebus 4" started
info: Rebus.RabbitMq.RabbitMqTransport[0] Successfully initialized consumer for "dev-workflow-services-worker6767dc84b4dttx-workflow-management-events"
info: Elsa.Runtime.StartupRunner[0] Running startup task ContinueRunningWorkflows
```

**Comment by sfmskywalker** [comment]
So the application just gets stuck, no exceptions or anything? Can you reproduce this on your local machine (using a copy of the DB)?

**Comment by johnwc** [comment]
Yes, I have actually seen this a few times in local development as well. What gets it fixed is me going into the `WorkflowInstances` table and delete everything, or just the one I know was running. I exported the `WorkflowInstances` table before reporting this and clearing the rows in case you wanted me to look at anything.

**Comment by johnwc** [comment]
So I finally was forced to track this down today. As our K8s servers had network issues and caused the pods to restart continually until it could access our container registry. It seems what is going on here is that when a server starts and calls to acquire a lock for [ContinueRunningWorkflows](https://github.com/elsa-workflows/elsa-core/blob/master/src/core/Elsa.Core/StartupTasks/ContinueRunningWorkflows.cs#L50), the [default timeout](https://github.com/elsa-workflows/elsa-core/blob/master/src/core/Elsa.Core/Options/ElsaOptions.cs#L62) is 1 hour. Does this `ContinueRunningWorkflows` expect to take potentially an hour to run, or can this be adjusted to something reasonable for this specific task?

**Comment by johnwc** [comment]
Scratch that, I now understand what is going on. One of the server only nodes is in the middle of starting all the workflows back up, and the dashboard+server node we use for UI access is blocked starting the web server until it finishes executing `ContinueRunningWorkflows`. Does this need to be started in the background maybe, so ASP.net can fully start without waiting on it to complete?

**Comment by sfmskywalker** [comment]
Yeah exactly. At the time, I considered doing it in the background and not block startup, but I was worried what would happen if requests came in looking for workflow instances that hadn't restarted yet. But I can't say for sure if it would pose an actual issue. IF you want to try and see what happens, you could change the `ContinueRunningWorkflows` implementation to be a `BackgroundService` so that it's not part of the `StartupRunner` service which is run from `StartupRunnerHostedService`, which is blocking startup until it finishes. Alternatively, you could try and change `StartupRunnerHostedService` itself to be a background service instead. The only potential issue I can see right now is the migration runner - if the migrations haven't run yet while another startup tasks access the DB, it might break. Then again, this would only happen if your DB schema isn't up to date with the latest migrations.
